src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -372,7 +428,15 @@ private void LoadEntryAt(Stream fs, long position, out string id, out int byteOf
 
             public string[] GetTimeZoneIds()
             {
-                return _ids;
+                List<string> nonBackwardsTZIDs = new List<string>();
```

Comment:
nit: this code is ok but if you want to optimize more to avoid allocating the list, you can count the number of the filtered out  zones and then allocate the returned string[] with exact size. 

I'll leave this up to you if you want to optimize it or not.

Reply:
Thanks for the suggestion! I looped through the `_isBackwards` array one extra time to count the number to allocate. It seems a bit weird to loop twice, but having a counter for the `nonBackwardsTZIDs` in the class would be allocating for an `int` for the entire lifetime of the class, and thats worse than just the lifetime of `GetTimeZoneIds` right?

Out of curiosity, what specifically are we avoiding in allocating the list? I am seeing that a list allocates space for a 2<sup>n</sup> array where its doubled each time it runs out of space. Are we avoiding that worst case 2<sup>n-1</sup>-1 space allocation or are we mainly avoiding the extra variables being allocated through the `List<T>` object?

Reply:
On mobile it's a very good idea to avoid any unnecessary allocations (on desktop/server too, but less critical) as it not only conserves resources but it also puts a lower pressure on the GC by generating less objects to scan.